spec/models/registration_spec.rb

```diff
@@ -1331,4 +1368,121 @@
+      expect(reg.paid_entry_fees.cents).to eq(1000)
+    end
+
+    it 'correctly sums multiple payments' do
+      create(:registration_payment, registration: reg)
+      expect(reg.reload.paid_entry_fees.cents).to eq(2000)
+    end
+
+    it 'returns 0 for a refunded payment' do
+      create(:registration_payment, :refund, registration: reg)
+      expect(reg.reload.paid_entry_fees.cents).to eq(0)
+    end
+
+    it 'returns net amount of multiple payments/refunds' do
+      create_list(:registration_payment, 2, registration: reg)
+      create_list(:registration_payment, 2, :refund, registration: reg)
+      expect(reg.reload.paid_entry_fees.cents).to eq(1000)
+    end
+
+    it 'correctly calculates a partial refund' do
+      create(:registration_payment, :refund, registration: reg, amount_lowest_denomination: -500)
+      expect(reg.reload.paid_entry_fees.cents).to eq(500)
+    end
+
+    it 'only considers captured payments' do
+      create(:registration_payment, registration: reg, is_completed: false)
+      expect(reg.reload.paid_entry_fees.cents).to eq(1000)
+    end
+  end
+
+  describe 'last_payment methods' do
+    before do
+      @expected_pmt = create(
+        :registration_payment, registration: registration, created_at: Time.now.utc-4, updated_at: Time.now.utc-4, paid_at: Time.now.utc-1
+      )
+      @other1 = create(
+        :registration_payment, registration: registration, created_at: Time.now.utc-3, updated_at: Time.now.utc-3, paid_at: Time.now.utc-3
+      )
+      @other2 = create(
+        :registration_payment, registration: registration, created_at: Time.now.utc-2, updated_at: Time.now.utc-2, paid_at: Time.now.utc-2
+      )
+    end
+
+    it 'paid_at for first payment is after all timestamps for other payments' do
+      expect(@expected_pmt.paid_at).to be > @other2.created_at
+      expect(@expected_pmt.paid_at).to be > @other2.updated_at
+      expect(@expected_pmt.paid_at).to be > @other2.paid_at
+      expect(@expected_pmt.paid_at).to be > @other1.created_at
+      expect(@expected_pmt.paid_at).to be > @other1.updated_at
+      expect(@expected_pmt.paid_at).to be > @other1.paid_at
+    end
+
+    it 'non-paid_at timestamps for first payment are before all timestamps for other payments' do
+      expect(@expected_pmt.created_at).to be < @other2.created_at
+      expect(@expected_pmt.updated_at).to be < @other2.created_at
+      expect(@expected_pmt.created_at).to be < @other2.updated_at
+      expect(@expected_pmt.updated_at).to be < @other2.updated_at
+      expect(@expected_pmt.created_at).to be < @other1.created_at
+      expect(@expected_pmt.updated_at).to be < @other1.created_at
+      expect(@expected_pmt.created_at).to be < @other1.updated_at
+      expect(@expected_pmt.updated_at).to be < @other1.updated_at
+    end
+
+    describe '#last_payment' do
+      context 'payments are loaded' do
+        before do
+          registration.registration_payments.load
```

Comment:
What is this?

Reply:
I'm testing the behaviour of `last_payment`, which has branching logic based on whether or not payments are loaded into memory: 
```ruby
def last_payment
    if registration_payments.loaded?
      registration_payments.completed.max_by(&:paid_at)
    else
      registration_payments.completed.order(:paid_at).last
    end
  end
```